chrome/content/zotero/xpcom/utilities_internal.js

```diff
@@ -806,6 +806,80 @@ Zotero.Utilities.Internal = {
 		parts.push(isbn.charAt(isbn.length-1)); // Check digit
 		
 		return parts.join('-');
+	},
+	
+
+	/**
+	 * Create a libraryOrCollection DOM tree to place in <menupopup> element.
+	 * If has no children, returns a <menuitem> element, otherwise <menu>.
+	 * 
+	 * @param {Library/Collection} libraryOrCollection
+	 * @param {function} clickAction function to execute on clicking the menuitem.
+	 * 		Receives the event and libraryOrCollection for given item.
+	 * 
+	 * @return {Node<menuitem>/Node<menu>}
+	 */
+	createMenuForTarget: function(libraryOrCollection, clickAction) {
+		var win = Services.wm.getMostRecentWindow("navigator:browser");
```

Comment:
Should just pass the document here — don't want to use windows. (And seems like the createElementNS came back below, which shouldn't be necessary.)